### ISSUE TITLE
TypeError: Cannot read property 'syntax' of undefined with --fix

### ISSUE THREAD

**Issue by dptoot** [stylelint/stylelint#2627]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where if the --fix flag is used as a CLI argument then the following error results

```
> husky - npm run -s prepush
> husky - node v6.10.0

TypeError: Cannot read property 'syntax' of undefined
    at stylelint._lintSource.then.postcssResult (/data/work/axl/node_modules/stylelint/lib/standalone.js:156:74)
```

Removal of the --fix flag causes stylelint to work as expected

> Which rule, if any, is this issue related to?

n/a

> What CSS is needed to reproduce this issue?
n/a

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "extends": "stylelint-config-standard",
  "rules": {
    "indentation": 4,
  }
}
```

> Which version of stylelint are you using?

└── stylelint@7.11.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

"lint:css": "stylelint \"packages/**/*.scss\" --syntax scss --fix",

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

I am linting SCSS but I don't feel it is the issue

> What did you expect to happen?

No errors :)

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
> husky - npm run -s prepush
> husky - node v6.10.0

TypeError: Cannot read property 'syntax' of undefined
    at stylelint._lintSource.then.postcssResult (/data/work/axl/node_modules/stylelint/lib/standalone.js:156:74)
```

**Comment by evilebottnawi** [comment]
Seems related to https://github.com/stylelint/stylelint/issues/2619

**Comment by evilebottnawi** [comment]
@dptoot weird, locally all work fine, can your provide test minimal reproducible repo?

**Comment by hudochenkov** [comment]
I can't reproduce it either with fresh stylelint install.

@dptoot @jitendravyas could you, please, show us your installed modules? Run `npm list` in directory where stylelint is installed and copy at least stylelint part (with all its dependencies).

And test minimal reproducible repo would be also much appreciated.

**Comment by jitendravyas** [comment]
I have three question before. 

When we use Stylelint CLI do we need to specificy the path of stylelintrc in cli command? 

Can we auto-fix a rule using cli without stylelintrc?

When we use CLI do I need to give absolute path of folder of my scss files or relative path will do?

**Comment by jeddy3** [comment]
These are all answered in the docs:

> When we use Stylelint CLI do we need to specificy the path of stylelintrc in cli command?

[Loading the configuration object](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#loading-the-configuration-object).

> Can we auto-fix a rule using cli without stylelintrc?

[Configuration](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#configuration) - "The linter _expects a configuration object_. You can either craft your own config or extend an existing one.". A configuration object tells stylelint what you want fixing and in _what way you_ want it fixed.

> When we use CLI do I need to give absolute path of folder of my scss files or relative path will do?

[CLI examples](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/cli.md#examples)

**Comment by jeddy3** [comment]
Closing. Please feel free to reopen when you're able to create a reproducible test case.

**Comment by subzey** [comment]
@jeddy3 I have no test case, but instead I have a stack trace, hope it's fine too.

Please look at the https://github.com/stylelint/stylelint/blob/master/lib/standalone.js#L147-L156. `_lintSource` is called and then `.opts.syntax` is read from the Promise resolution.

![2017-06-19-184138_1280x1024_scrot](https://user-images.githubusercontent.com/810582/27293409-01971e50-551f-11e7-931b-6d9d41790038.png)

At the same time in https://github.com/stylelint/stylelint/blob/master/lib/lintSource.js#L45-L50 `lintSource` can return/resolve an object *without* `opts` property if file is ignored.

![2017-06-19-184126_1280x1024_scrot](https://user-images.githubusercontent.com/810582/27293404-ffa77446-551e-11e7-819a-1bd61d838798.png)

**Comment by subzey** [comment]
Here's a setup that reproduces the error:

**stylelint v7.11.1**

.stylelintrc.js
```js
module.exports.rules = {}; // No rules needed
```
.stylelintignore
```
style.css
```
style.css
```css
/* Can be empty */
```

Running `$(npm bin)/stylelint *.css --fix` throws.

**Comment by hudochenkov** [comment]
@subzey thank you for digging into the problem and doing research!

**Comment by hudochenkov** [comment]
Done in https://github.com/stylelint/stylelint/pull/2652

**Comment by jeddy3** [comment]
@subzey Thanks for your help! This will be released in `7.12`.

**Comment by jitendravyas** [comment]
I hope finally I will be able to use `--fix`. Thanks.

**Comment by jitendravyas** [comment]
I get this error now with `7.12.0`

```
❯ stylelint "app/styles/**/*.scss" --fix
TypeError: Cannot read property 'syntax' of undefined
    at stylelint._lintSource.then.postcssResult (/Users/jitendravyas/.nvm/versions/node/v6.10.0/lib/node_modules/stylelint/lib/standalone.js:156:74)
```
